index.js

```diff
@@ -2613,6 +2613,42 @@
   }
   S.ifElse = def('ifElse', {}, [Pred(a), Fn(a, b), Fn(a, b), a, b], ifElse);
 
+  //# when :: (a -> Boolean) -> (a -> b) -> a -> b
```

Comment:
in ramda type is:
`(a → Boolean) → (a → a) → a → a`
So i think it we should go this way or:
`(a → Boolean) → (a → b) → a → Either a b`